src/templates.ts

```diff
@@ -87,6 +110,30 @@ export const downloadTemplates =
     }
   }
 
+const enrichWithPrePendingHeader =
+  (mustacheRenderedContent: string, template: Template, codeowners: string) =>
+  (log: Logger): string => {
+    // TODO Check if file extensions fits together
+    const templateExtension = template.destinationPath.split('.').pop() as string
+    if (!(templateExtension == 'yaml' || templateExtension == 'toml' || templateExtension == 'yml')) {
+      log.debug(`File extension: ${templateExtension} with not supported comments`)
+      return mustacheRenderedContent
+    }
+
+    const codeOwnersEntries = parse(codeowners)
+    const matchedWithCodeOwner = matchFile(template.sourcePath, codeOwnersEntries)
+
+    let prependingHeader: string
+
+    if (typeof process.env.PREPENDING_HEADER_TEMPLATE != 'undefined' && process.env.PREPENDING_HEADER_TEMPLATE) {
+      prependingHeader = `${process.env.PREPENDING_HEADER_TEMPLATE + matchedWithCodeOwner?.owners}\n\n`
+    } else {
+      prependingHeader = `#OWNER: ${matchedWithCodeOwner?.owners}\n\n`
+    }
+
+    return prependingHeader + mustacheRenderedContent
```

Comment:
I'm a bit allergic to side-effects. I'd prefer if we just return the result right away to not keep track of the `let` assignment above 😄 

```suggestion
    if (process.env.PREPENDING_HEADER_TEMPLATE) {
      return `${process.env.PREPENDING_HEADER_TEMPLATE + matchedWithCodeOwner?.owners}\n\n${mustacheRenderedContent}`
    } else {
      return `#OWNER: ${matchedWithCodeOwner?.owners}\n\n${mustacheRenderedContent}`
    }
```

WDYT?

Reply:
`if (process.env.PREPENDING_HEADER_TEMPLATE)` 

this is the equivalent of 

`if (typeof process.env.PREPENDING_HEADER_TEMPLATE != 'undefined' && process.env.PREPENDING_HEADER_TEMPLAT`

?

Reply:
Also see: https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
Not super nice, but it is what it is. You basically want to always use `===` to check the existence of something and `!` to check for something not being present to be somewhat safe